Add banner and accent color to user and some fixes/improvements

INikonI opened the pull request:
+ Resolve #74 
+ Mini refactor of color class and add to them implicit operators for `uint` and const with max decimal value
+ Some fixes on `Test.Unit`

INikonI committed:
Add banner and accent color to user and some fixes

INikonI committed:
Fix

INikonI committed:
Fix!

quinchs commented on src/Discord.Net.Core/Entities/Users/IUser.cs:
```diff
@@ -34,6 +46,16 @@ public interface IUser : ISnowflakeEntity, IMentionable, IPresence
+        /// <returns>
+        ///     A string representing the user's avatar URL; <c>null</c> if the user does not have an banner in place.
+        /// </returns>
+        string GetBannerUrl(ImageFormat format = ImageFormat.Auto, ushort size = 128);
```
since banners are bigger images, shouldn't the default size be bigger?

also is the size of the image the width or height?

INikonI replied:
> since banners are bigger images, shouldn't the default size be bigger?

increased in b8b92b7be8470a3aad8d7bc25f41028be9ce2d12, but 128 is default width for images in discord

quinchs commented on src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:
```diff
@@ -53,6 +53,20 @@ public override string AvatarId
+        public override string BannerId
+        {
+            get => GuildUser.BannerId;
+            internal set => GuildUser.BannerId = value;
```
why add setters if they are not used?

INikonI replied:
idk, `AvatarId` property too has setter

INikonI replied:
Because it implementation of interfaces/classes, if specifically then implementation of `SocketUser`.

quinchs replied:
Any property like that will be updated in the class that owns it, therfor we dont need to modify any properties out of sync with that guild user. Could you please remove it

INikonI replied:
ok

INikonI replied:
but how 🤔

INikonI replied:
This class created by you, and inherits `SocketUser`. `AvatarID` and other props also has setters, because its implementation of `SocketUser`.

quinchs replied:
just dont override it...

quinchs replied:
nvm its good ig.

quinchs commented on src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs:
```diff
@@ -24,6 +24,13 @@ public class SocketWebhookUser : SocketUser, IWebhookUser
+        public override string BannerId { get; internal set; }
+
+        /// <inheritdoc />
+        public override Color? AccentColor { get; internal set; }
```
webhooks cannot have banner/accent colors, please make it throw a NotSupportedException

INikonI replied:
its just `null` for bots and webhooks? 🤔

INikonI replied:
Resolved in abcb4e7a9a524d79d54ff980629ffebb36ae54f7

INikonI committed:
increase size of user banners to 256

INikonI committed:
Some changes and mini refactor of color class

INikonI committed:
add constant maxDecimalValue to color and checks with exceptions

INikonI committed:
add `NotSupportedException` for `BannerId` and `AccentColor` in `SocketWebhookUser`

INikonI committed:
Merge branch 'Discord-Net-Labs:release/2.x' into release/2.x

quinchs commented on src/Discord.Net.Core/Entities/Roles/Color.cs:
```diff
@@ -10,68 +10,70 @@ namespace Discord
+        public static readonly Color Magenta = new(0xE91E63);
         /// <summary> Gets the dark magenta color value. </summary>
         /// <returns> A color struct with the hex value of <see href="http://www.color-hex.com/color/AD1457">AD1457</see>.</returns>
-        public static readonly Color DarkMagenta = new Color(0xAD1457);
```
Why did you change the constructor?

INikonI replied:
Redundant type specification

quinchs commented:
Please fix the conflicts in `test/Discord.Net.Tests.Unit/GuildPermissionsTests.cs`

INikonI committed:
Merge branch 'release/2.x' into release/2.x

INikonI committed:
Update ComponentBuilder.cs

- `MaxLabelLength` from `ComponentBuilder` moved to `ButtonBuilder`
- Added `MaxLabelLength` for `SelectMenuOptionBuilder`
- Changed `MaxDescriptionLength` to 100

INikonI commented:
Reason of 2370f62c6fc0d99e22aeae5a35ae06775212bdf9 commit 
![image](https://user-images.githubusercontent.com/47792796/128607074-622d3935-2cca-4a12-899d-a8246b0fa0b6.png)
